prune_shard_list.c

```diff
@@ -239,6 +249,51 @@ MakeOpExpression(Var *variable, int16 strategyNumber)
 
 
 /*
+ * LookupOperatorByType is a wrapper around GetOperatorByType that uses a cache
+ * to avoid multiple lookups of operators by their types.
+ */
+static Oid
+LookupOperatorByType(Oid typeId, Oid accessMethodId, int16 strategyNumber)
+{
+	OperatorTypeCacheEntry *matchingCacheEntry = NULL;
+	ListCell *cacheEntryCell = NULL;
+
+	/* search the cache */
+	foreach(cacheEntryCell, OperatorTypeCache)
+	{
+		OperatorTypeCacheEntry *cacheEntry = lfirst(cacheEntryCell);
+
+		if (cacheEntry->typeId == typeId && cacheEntry->accessMethodId == accessMethodId
```

Comment:
Style-wise, I think we need explicit grouping here because there are more than two expressions. Can you change to:

```c
if ((cacheEntry->typeId == typeId) &&
	(cacheEntry->accessMethodId == accessMethodId) &&
	(cacheEntry->strategyNumber == strategyNumber))
{
```
It'll be more readable with the parallel `cacheEntry` variables lined up like that.